rest/config.go

```diff
@@ -1225,6 +1225,24 @@ func (sc *ServerContext) applyConfigs(dbNameConfigs map[string]DatabaseConfig) (
 
 // _applyConfig loads the given database, failFast=true will not attempt to retry connecting/loading
 func (sc *ServerContext) _applyConfig(cnf DatabaseConfig, failFast bool) (applied bool, err error) {
+	// 3.0.0 doesn't write a SGVersion, but everything else will
+	configSGVersionStr := "3.0.0"
+	if cnf.SGVersion != "" {
+		configSGVersionStr = cnf.SGVersion
+	}
+
+	configSGVersion, err := base.NewComparableVersionFromString(configSGVersionStr)
+	if err != nil {
+		return false, err
+	}
+
+	// Skip applying if the config is from a newer SG version than this node
+	nodeSGVersion := base.ProductVersion
+	if nodeSGVersion.Less(configSGVersion) {
```

Comment:
To confirm, this is not just the update case, but also means that databases updated by a newer version of Sync Gateway will not be loaded at all if the older Sync Gateway node is restarted, correct?

This seems like it prevents downgrade scenarios (customer hits an issue in 3.1.1 after updating the db config, can't roll back SG to 3.1.0).  Any thoughts on how customers would deal with that situation?

Reply:
Ugh, you're right... This case was intended to just avoid the situation where a mixed-version cluster is running and receving newer config updates from server without falling over...

But yeah, on a restart/downgrade the node would still end up using this code-path to pull their config from the server on startup.

I think we can differentiate between initial startup and the continuous config updates, and try to apply the version to the node regardless of version checking in the initial startup case.

Reply:
Had to rebase, but the last commit contains all of the changes made to support this downgrade use-case (along with test coverage)